src/rcx/doc/model_file.README

```diff
@@ -0,0 +1,137 @@
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+# Author: Dimitris K. Fotakis
+# Version: v1.0
+# Date: 11/08/2024
+
+NOTE: this is NOT a comprehensive File Format description, it is an aid to read a model file
+
+
+# ------------------------------------------------------------
+# 			Model File Structure
+# ------------------------------------------------------------
+
+
+# ------------------------------------------------------------
+	A. Header Section
+# ------------------------------------------------------------
+
+
+	Following line are not Used:
+		Extraction Rules for rcx
+		Version 1.2
+		DIAGMODEL ON
+
+LayerCount <num> : number of metal levels
+	example: LayerCount 6
+
+
+DensityRate <corner_cnt> <corner indices list> : number of process corners
+	example: DensityRate 3  0 1 2
+
+Corners <corner_cnt> <corner_name_list> : names of process corners for documentation purposes
+	example: Corners 3 :  max typ min
+
+COMMENT : <any comment in one line> 
+	example: COMMENT : v1 
+
+
+# ------------------------------------------------------------
+	B. Procesll Corner Start
+# ------------------------------------------------------------
+
+DensityModel <process_corner_index>
+	example: DensityModel 0
+
+# ------------------------------------------------------------
+	C. Extraction Pattern Model
+# ------------------------------------------------------------
+
+Metal <num> <model_name> 
+	model_name can be:
+		RESOVER - Distance based Resistance
+		OVER  - metal over sub or other metal
+		OVER0  - metal over sub or other metal - 1-wire model
+		OVER1  - metal over sub or other metal - 2-wire model
+		UNDER - metal over metal under Substrate
+		UNDER0 - metal over metal under Substrate - 1-wire model
+		UNDER1 - metal over metal under Substrate - 2-wire model
+		OVERUNDER - metal over metal under metal
```

Comment:
What about OVERUNDER0 & OVERUNDER1 which are in `extRCModel::spotModelsInRules`?